classes/class-p4-metabox-register.php

```diff
@@ -0,0 +1,420 @@
+				'id'         => $this->prefix . 'campaign_name',
+				'type'       => 'select',
+				'options'    => $campaign_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$basket_options = [
+			'not set'              => __( '- Select Basket -', 'planet4-master-theme-backend' ),
+			'Forests'              => 'Forests',
+			'Oceans'               => 'Oceans',
+			'Good Life'            => 'Good Life',
+			'Food'                 => 'Food',
+			'Climate & Energy'     => 'Climate & Energy',
+			'Oil'                  => 'Oil',
+			'Political & Business' => 'Political & Business',
+		];
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Basket Name', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'basket_name',
+				'type'       => 'select',
+				'options'    => $basket_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$scope_options = [
+			'not set'  => __( '- Select Scope -', 'planet4-master-theme-backend' ),
+			'Global'   => 'Global',
+			'National' => 'National',
+		];
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Scope', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'scope',
+				'type'       => 'select',
+				'options'    => $scope_options,
+				'attributes' => [
+					'data-validation' => 'required',
+				],
+			]
+		);
+
+		$p4_campaign_fields->add_field(
+			[
+				'name'       => __( 'Department', 'planet4-master-theme-backend' ),
+				'id'         => $this->prefix . 'department',
+				'type'       => 'text_medium',
+				'attributes' => [
+					'placeholder' => __( 'Add Department', 'planet4-master-theme-backend' ),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Populate an associative array with all the children of the ACT page
+	 *
+	 * @return array
+	 */
+	public function populate_act_page_children_options() {
+		$parent_act_id = planet4_get_option( 'act_page' );
+		$options       = [];
+
+		if ( 0 !== absint( $parent_act_id ) ) {
+			$take_action_pages_args = [
+				'post_type'        => 'page',
+				'post_parent'      => $parent_act_id,
+				'post_status'      => 'publish',
+				'orderby'          => 'menu_order',
+				'order'            => 'ASC',
+				'suppress_filters' => false,
+				'numberposts'      => self::MAX_TAKE_ACTION_PAGES,
+			];
+
+			$posts = get_posts( $take_action_pages_args );
+			foreach ( $posts as $post ) {
+				$options[ $post->ID ] = $post->post_title;
+			}
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Validate CMB2 fields that have the 'data-validation' attribute set to 'required'.
+	 *
+	 * @param int  $post_id The ID of the current Post.
+	 * @param CMB2 $cmb The CMB2 object.
+	 */
+	public function cmb2_after_form_do_js_validation( $post_id, $cmb ) {
+		static $added = false;
+
+		// Only add this to the page once (not for every metabox).
+		if ( $added ) {
+			return;
+		}
+
+		$added = true;
```

Comment:
Could use [wp_script_is](https://codex.wordpress.org/Function_Reference/wp_script_is) to check if it is already loaded
`if ( ! wp_script_is( 'cmb2_form_validation' ) ) {` but if that works as expected then I guess its ok.

Reply:
If we don't have [this](https://github.com/greenpeace/planet4-master-theme/pull/770/files#diff-03bce058cc5b9903fb8e557c6e4c4ff0R413) check then it would be the good fit.